packages/canary/src/components/navbar-user.tsx

```diff
@@ -1,25 +1,55 @@
 import React from 'react'
 import { getInitials } from '@/utils/StringUtils'
-import { Avatar, AvatarFallback, AvatarImage } from '..'
+import {
+  Avatar,
+  AvatarFallback,
+  AvatarImage,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger
+} from '..'
+
+interface items {
+  element: React.ReactNode
+  key: number
+}
 
 const Root: React.FC<{
   username?: string
   isAdmin?: boolean
   url?: string
-}> = ({ username, isAdmin, url }) => {
+  menuItems?: items[]
+}> = ({ username, isAdmin, url, menuItems }) => {
   return (
-    <div className="grid grid-rows-2 grid-cols-[auto_1fr] gap-x-3 items-center justify-start cursor-pointer">
-      <div className="col-start-1 row-span-2">
-        <Avatar>
-          {url && <AvatarImage src={url} alt="user" />}
-          {username && <AvatarFallback>{getInitials(username)}</AvatarFallback>}
-        </Avatar>
-      </div>
-      <p className="col-start-2 row-start-1 text-xs text-primary font-medium">{username}</p>
-      <p className="col-start-2 row-start-2 text-xs font-normal text-tertiary-background">
-        {isAdmin ? 'Admin' : 'Account Member'}
-      </p>
-    </div>
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <div className="group relative grid grid-rows-2 grid-cols-[auto_1fr] gap-x-3 items-center justify-start cursor-pointer">
+          <div className="absolute -inset-2 rounded-md group-hover:bg-primary/5 ease-in-out duration-100" />
+          <div className="col-start-1 row-span-2">
+            <Avatar className="rounded-md overflow-hidden">
+              {url && <AvatarImage src={url} alt="user" />}
+              {username && <AvatarFallback>{getInitials(username)}</AvatarFallback>}
+            </Avatar>
+          </div>
+          <p className="col-start-2 row-start-1 text-xs text-primary font-medium">{username}</p>
+          <p className="col-start-2 row-start-2 text-xs font-normal text-tertiary-background">
+            {isAdmin ? 'Admin' : 'Account Member'}
```

Comment:
there can be other roles potentially. let's not assume just two. it'd be better to pass-in the role as a prop and display it as-is.

Reply:
Currently the API gives us the user where admin is a boolean. 

```
export interface TypesUser {
    admin?: boolean;
    blocked?: boolean;
    created?: number;
    display_name?: string;
    email?: string;
    uid?: string;
    updated?: number;
}
```

Hence the current implementation checks for `admin` and if not assigns `Account Member`. I think we should keep this for now until we get more roles added because this aligns well with the backend for now.

Reply:
Agreed, that might be true for the gitness backend, but if we decide to re-use this component for Harness, this won't apply. Ideally I'd like Playground components to be independent of Gitness API specifics.
But in either case, this is not a blocker.